### PR TITLE
Added saveMeal function and onClick event for saving the recipe

### PR DESCRIPTION
**Pull request by bendered2332**
Description of the Project:
This application is a great idea and it reminds me a lot of Allrecipes as it provides a lot of recipes based on a few keywords.

- The README.md shows what the final product will look like and the group is on a great path to reaching that goal. 
- The project easily loaded into my Android Studio and built without any errors which were great!
-  Their project data is from a reputable source and is full-on entries.

Was the program in UC Github on time? 
- Yes, the project was on Github in time. 

Is the program documented/commented well enough for you to understand? 
The project was very well documented and when anything was unclear there were comments describing the functionality of the function and or class.

Does the program compile?
Yes, the program compiles perfectly.

The rationale behind your changes.
I added some save functionality to the save button on click event. This would allow the group to save their meal to firebase if they wanted to set up their saveMeal function that way. I also created the saveMeal function.

Links to three specific commits that you made in your own Group's Github repository.
https://github.com/pranavm7/UniConnect/commit/0273cdd726035f3191293d1bd5d6f23a80b7cd6e 
https://github.com/pranavm7/UniConnect/commit/68587f5a7580b84a406e42fffe59e35d37cff22e
https://github.com/pranavm7/UniConnect/commit/b4eeda57feb3cbdf746fff918f3a874cbbdab1a2

### CODE REVIEW

**Comment by bendered2332** [comment]
Next, list three specific technical concepts that you learned from this code review. Copy and past samples to demonstrate your point.

1. I learned how to add the camera to the android application and then save the photos to firebase.  They also followed your videos from module 10. https://uc.instructure.com/courses/1514344/pages/module-10-content-and-activities?module_item_id=66029371 
2. I also learned how to change the MaterialTheme color of the entire application
3. I learned through their mainActivity.kt how to make a text field with a dropdown menu.

File: app/src/main/java/com/cookit/MainViewModel.kt [hunk 1]
```diff
@@ -22,4 +23,8 @@ class MainViewModel(var recipeService: IRecipeService = RecipeService()) : ViewM
             recipes.postValue(innerRecipes)
         }
     }
+
+    fun saveRecipe(meal: Meal) {
+
+    }
 }
```
Comment (discospiff) [inline app/src/main/java/com/cookit/MainViewModel.kt:30]: This looks like an incomplete feature.
Are you using these anywhere? 
If not in use, remove them.  Keep the codebase as small as possible to do the job you want to do.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.

**Comment by discospiff** [comment]
This branch introduces new, and possibly incomplete, functionality, instead of reducing technical debt of existing functionality.  New functionality should be kept in a separate branch until it is complete and meets the team's Definition of Done.  Thus, if the team would like these features, they can keep the branch open and continue to work on it, and then merge once it meets the Definition of Done.  Otherwise, if they do not want these changes, they can abandon/delete this branch and not merge it.

**Comment by ccbrowndev** [comment]
Closing because we already have a branch where we are working on a save to firebase feature, but that branch has much more progress

**Pull request closed without merging**